0000-new_pulse_drawer.md

```diff
@@ -0,0 +1,274 @@
+  return matplotlib.Figure
+```
+
+```python
+def pulse_drawer_core(program: SamplePulse, ...):
+  # draw schedule figure
+  return matplotlib.Figure
+```
+
+The `schedule_drawer_core` is used to draw pulse `Schedule` while the `pulse_drawer_core` is used to draw `SamplePulse`.
+Since a pulse `Schedule` has more complecated data structure, the `schedule_drawer_core` consists of several elements to extract a set of objects to draw.
+
+The class `EventsOutputChannels` is an extention of existing `EventsOutputChannels` class, but some method names are updated to reflect the recent change of pulse syntax. This class is defined for each `Channel` to draw and the class makes a collection of instructions tagged by the time. The class structure looks like:
+
+```python
+class EventsOutputChannels:
+  def __init__(self, schedule: ScheduleComponent, channel: Channel):
+    self._pulse = []
+    self._phase = []
+    self._frequency = []
+    self._snapshot = []
+    self._conditional = []
+
+    self._parse_schedule(schedule, channel)
+
+  def _parse_schedule(self, schedule, channel):
+    # filter schedule by channel and make time table of instructions
+    pass
+
+  def waveform(self, t0, tf) -> List[Tuple[ndarray, str]]:
+    # create waveform from time t0 to tf and return it as list of ndarray
+    pass
+
+  @property
+  def max_amp(self):
+    # return max amplitude in the channel
+    pass
+
+  @property
+  def min_amp(self):
+    # return min amplitude in the channel
+    pass
+```
+
+The main purpose of this class to generate a list of complex `ndarray` accompanied with the pulse name from the given set of instruction in the channel.
+Note that pulse waveform should consider the phase instructions.
+This is done by calling `waveform` method.
+Each `ndarray` in the list represents a pulse envelope of each `Play` instruction.
+The interval between `Play` instructions should be padded by `Delay` and this appears as long zero array with the length of the interval in terms of system cycle time (not actual pulse time).
+This is because we need to truncate long pulses or long delays based on `truncate` preference in the style sheet.
+If we output the entire waveform as a single data array it is very hard to recover a length of each pulse.
+The method `max_amp` and `min_amp` are used to determine the autoscaling value of the channel.
+
+First the `schedule_drawer_core` initialize `EventsOutputChannels` for all channels specified by the user input and the style sheet preference and keep them as a python `dict`. Then, those channels are rearanged as a `list` according to the `layout` option in the style sheet. If `'layout': 'channel-type-wise'` is specified, the program collects all `DriveChannel`s in the dictionary and then collects `ControlChannel`s, ... , so that channels of the same type are shown in one group.
+
+In `utils.py` several helper functions are defined to extract information and coordinate to show from the `EventsOutputChannels`.
```

Comment:
Does this need to be in a utils module? Could it just live in one of the other modules?